### ISSUE TITLE
Unnecessary property generated when MultiValueMap<> used as request parameter

### ISSUE THREAD

**Issue by mirza-ha** [vojtechhabarta/typescript-generator#577]
Hello.
I'm trying to generate typescript for my spring boot project, and everything works fine (except issue #576 that I already added), but I'm having a problem when I use MultiValueMap as request parameter.

If I declare my controller method, as for example:
``` java
@GetMapping
search(@RequestParam MultiValueMap<String, String> params)
```

what I'm getting as generated code is:
``` typescript
search(queryParams: { params: { [index: string]: string }; }) {
    return this.httpClient.request({ method: "GET", url: uriEncoding`xy`, queryParams: queryParams }); 
}
```

so "params" is serialised as additional object, and backend is not able to deserialise it as map.

When I change generated code to:
``` typescript
search(queryParams: { [index: string]: string }) {
     return this.httpClient.request({ method: "GET", url: uriEncoding`xy`, queryParams: queryParams }); 
}
```
or add "queryParams: queryParams.params" at the end, everything works fine.

Can this additional generated property be excluded somehow?

**Comment by vojtechhabarta** [comment]
I committed support for this case. Generated parameter is `queryParams?: { [index: string]: any }` which allows user to pass single values as well as multiple values for specified keys and it allows any TypeScript type which is convertible to string value by HTTP client.

**Comment by mirza-ha** [comment]
Thank you very much. Looking forward to the release.

**Comment by vojtechhabarta** [comment]
Released in [v2.28.785](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.28.785).
